src/web/mjs/connectors/ComicK.mjs

```diff
@@ -20,75 +21,87 @@ export default class ComicK extends Connector {
 
     async _getMangaFromURI(uri) {
         const data = await this._getEmbeddedJSON(uri);
-        return new Manga(this, data.comic.id, data.comic.title.trim());
+        return new Manga(this, data.comic.slug, data.comic.title.trim());
     }
 
     async _getMangas() {
         let mangaList = [];
-        for (let page = 0, run = true; run; page++) {
+        for (let page = 1, run = true; run; page++) {
             const mangas = await this._getMangasFromPage(page);
             mangas.length > 0 ? mangaList.push(...mangas) : run = false;
         }
         return mangaList;
     }
 
     async _getMangasFromPage(page) {
-        const uri = new URL('/api/get_newest_chapters?page=' + page, this.url);
+        const uri = new URL('/search?page=' + page, this.apiurl);
         const request = new Request(uri, this.requestOptions);
-        const data = await this.fetchJSON(request);
-        return !data.data ? [] : data.data.map(item => {
+        const data = await this.fetchJSONEx(request);
+        return data.message ? [] : data.map(item => {
             return {
-                id: item.md_comics.id,
-                title: item.md_comics.title.trim()
+                id: item.slug,
+                title: item.title.trim()
             };
         });
     }
 
     async _getChapters(manga) {
         let chapterList = [];
+        const comicId = await this._getComicId(manga);
         for (let page = 1, run = true; run; page++) {
-            const chapters = await this._getChaptersFromPage(manga, page);
+            const chapters = await this._getChaptersFromPage(comicId, page);
             chapters.length > 0 ? chapterList.push(...chapters) : run = false;
         }
         return chapterList;
     }
 
-    async _getChaptersFromPage(manga, page) {
-        const uri = new URL('/api/get_chapters', this.url);
-        uri.searchParams.set('comicid', manga.id);
-        uri.searchParams.set('page', page);
+    async _getChaptersFromPage(comicId, page) {
+        const uri = new URL(`/comic/${comicId}/chapter?page=${page}`, this.apiurl);
         const request = new Request(uri, this.requestOptions);
-        const data = await this.fetchJSON(request);
-        return data.data.chapters.map(item => {
-            let title = [];
+        const data = await this.fetchJSONEx(request);
+        return data.chapters.map(item => {
+            let title = '';
             if(item.vol) {
-                title.push('Vol.', item.vol);
+                title += `Volume ${item.vol} `;
```

Comment:
Downloaded chapters are identified by their name, when a new naming pattern is applied, user who have downloaded chapters will no longer see them as downloaded...

Reply:
fixed